clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -119,6 +123,11 @@ public Fetcher(ConsumerNetworkClient client,
      * @param cluster The current cluster metadata
      */
     public void initFetches(Cluster cluster) {
+        // Don't initiate fetch if we already have records available from previous fetchRequest
+        // This may happen when there is OffsetOutOfRange exception from previous fetchRequest
+        if (!records.isEmpty())
```

Comment:
Thanks for your comments. In order to keep data from ALL partitions of ALL `fetchResponse` which doesn't have `OffsetOutOfRangeException`, how about we do the following:

In the `initFetches()`, we only create `fetchRequest` for those partitions whose fetched position == consumed position. By doing so, `initFetch()` will only issue `fetchRequest` for partitions which caused `OffsetOutOfRangeException`; `poll(timeout)` will unblock upon `fetchResponse`; and the next successful `fetchedRecords()` will return data from all fetched partitions since the last successful `fetchedRecords()`.

Can you find anything problematic with this approach?